src/platform/src/main/java/org/geoserver/platform/resource/FileSystemResourceStore.java

```diff
@@ -429,7 +432,7 @@ public String toString() {
 
         @Override
         public boolean delete() {
-            return file.delete();
+            return file.exists() && Files.delete(file);
         }
```

Comment:
Not sure about this one, the functionality was isolated into the Resources class - specifically so implementors (like you) would have less work to do when implementing a Resource.

See: That functionality was already accounted for ...

The functionality is factored out into the Resources class here:

https://github.com/geoserver/geoserver/blob/master/src/platform/src/main/java/org/geoserver/platform/resource/Resources.java#L290

If you are going to change the method contract you will need to update the Resource.delete() javadoc to explain what you are about, and remove the Resources.delete( Resource ) method as it will now be useless.